src/services/goToDefinition.ts

```diff
@@ -197,15 +204,29 @@ namespace ts.GoToDefinition {
     }
 
     function getSymbol(node: Node, checker: TypeChecker): Symbol | undefined {
-        const symbol = checker.getSymbolAtLocation(node);
+        let symbol = checker.getSymbolAtLocation(node);
         // If this is an alias, and the request came at the declaration location
         // get the aliased symbol instead. This allows for goto def on an import e.g.
         //   import {A, B} from "mod";
         // to jump to the implementation directly.
-        if (symbol && symbol.flags & SymbolFlags.Alias && shouldSkipAlias(node, symbol.declarations[0])) {
-            const aliased = checker.getAliasedSymbol(symbol);
-            if (aliased.declarations) {
-                return aliased;
+        while (symbol) {
+            if (symbol.flags & SymbolFlags.Alias && shouldSkipAlias(node, symbol.declarations[0])) {
+                const aliased = checker.getAliasedSymbol(symbol);
+                if (!aliased.declarations) {

```

Comment:
when does this happen?

Reply:
This is just the converse of the previous code, but I think it could happen in certain error cases where the checker falls back to `unknownSymbol`. Not sure if that’s actually possible in `getAliasedSymbol`, but I feel like our assumption that symbols always have declarations is a frequent source of crashes, particularly in error-ridden code.